onadata/apps/api/management/commands/delete_users.py

```diff
@@ -0,0 +1,116 @@
+"""
+Delete users management command.
+"""
+import sys
+from django.contrib.auth.models import User
+from onadata.apps.logger.models import XForm
+from onadata.apps.logger.models import Instance
+from onadata.apps.logger.models import Project
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+
+def get_user_object_stats(username):  # pylint: disable=R0201
+    """
+    Get User information.
+    """
+    user = User.objects.get(username=username)
+    # Get the number of projects for this user
+    user_projects = Project.objects.filter(created_by=user).count()
+    # Get the number of forms
+    user_forms = XForm.objects.filter(user=user).count()
+    # Get the number of submissions
+    user_sumbissions = Instance.objects.filter(user=user).count()
+
+    result = {
+        'projects': user_projects,
+        'forms': user_forms,
+        'submissions': user_sumbissions
+    }
+
+    return result
+
+
+def inactivate_user(username):
+    """
+    Soft deletes the user termporarily.
+    """
+    try:
+        user = User.objects.get(username=username)
+        # set inactive status on user account
+        user.is_active = False
+        # append a timestamped suffix to the username
+        # to make the initial username available
+        soft_deletion_time = timezone.now()
+        deletion_suffix = soft_deletion_time.strftime('-deleted-at-%s')
+        user.username += deletion_suffix
+
+        user.save()
+        sys.stdout.write('User {} deleted successfully.'.format(username))
+        # confirm too that no user exists with provided email
+        try:
+            User.objects.get(email=user.email).delete()
```

Comment:
Why not use .filter() and check length ? What happens if the user wasn't deleted, should we notify the user of the command somehow ?

Reply:
Mhh actually... I might be misunderstanding this part of the code.. Is this deleting the user?

Reply:
Seems like line `48` soft deletes then this permanently deletes ?

Reply:
Yeah, that was my bad, I should be checking if an active user exists with the same user and inactivating those accounts. Not deleting them.
Changed this.